opm/simulators/flow/GenericOutputBlackoilModule.cpp

```diff
@@ -624,6 +628,38 @@ assignToSolution(data::Solution& sol)
         DataEntry{"UREA",     UnitSystem::measure::density,            cUrea_},
     };
 
+    // basically, for compositional, we can not use std::array for this.  We need to generate the ZMF1, ZMF2, and so on
+    // and also, we need to map these values.
+    // TODO: the following should go to a function
+    if (this->isCompositional_) {
+        auto compositionalEntries = std::vector<DataEntry>{};
+        {
+            // ZMF
+            for (int i = 0; i < numComponents; ++i) {
+                const std::string name = "ZMF" + std::to_string(i + 1);  // Generate ZMF1, ZMF2, ...
```

Comment:
You have `fmt::format()` in this file, so it's a lot better to write this as
```C++
const auto name = fmt::format("ZMF{}", i + 1);
```
instead.

Reply:
done fixing.